app/controllers/editions_controller.rb

```diff
@@ -343,7 +348,12 @@ def redirect_url
   end
 
   def make_govuk_url_relative(url = "")
-    url.sub(%r{^https?://(www\.)?gov\.uk/}, "/")
+    url.sub(%r{^(https?://)?(www\.)?gov\.uk/}, "/")
+  end
+
+  def validate_redirect(redirect_url)
+    regex = /(\/([a-z0-9]+-)*[a-z0-9]+)+/
+    redirect_url =~ regex
```

Comment:
I could be missing something but I'm not sure this is right:

```
irb(main):005:0> "/hello/world" =~ /(\/([a-z0-9]+-)*[a-z0-9]+)+/
=> 0
```

Should the final line be `redirect_url =~ regex == 0`?  The `redirect_url` is valid if the regex matches in position 0.

Reply:
I tested the previous version in integration and `redirect_url =~ regex` worked (I got the 'Redirect path is invalid' message when I entered 'redirect-to'). 
I will check `redirect_url =~ regex == 0`

Reply:
Actually it seems it was just me not understanding Ruby, 0 is truthy:

```
irb(main):015:0> if 0 || false
irb(main):016:1> "truthy"
irb(main):017:1> else
irb(main):018:1> "falsey"
irb(main):019:1> end
=> "truthy"
```

The regex is too lenient for what you're going for I think:

```
irb(main):021:0> "////hello/world-------" =~ /(\/([a-z0-9]+-)*[a-z0-9]+)+/
=> 3
```

But I'm not convinced that's a problem, it doesn't need to be terribly strict.